pkg/app/app.go

```diff
@@ -125,6 +126,33 @@ func defaultInputDirectory() configuration.DefaultValueFunction {
 	return callback
 }
 
+func defaultTempDirectory(engine workflow.Engine, config configuration.Configuration, logger *zerolog.Logger) configuration.DefaultValueFunction {
+	callback := func(existingValue interface{}) interface{} {
+		if existingValue != nil {
+			return existingValue
+		}
+		ri := engine.GetRuntimeInfo()
+		if ri == nil {
+			return false
+		}
+
+		version := ri.GetVersion()
+		if len(version) == 0 {
+			version = "0.0.0"
+		}
+
+		tmpDir := pkg_utils.GetTemporaryDirectory(config.GetString(configuration.CACHE_PATH), version)
+		err := pkg_utils.CreateAllDirectories(tmpDir, version)
+		if err != nil {
+			logger.Err(err)
+			return false
```

Comment:
Suggestion: Please always return the same type, in this case string. Also it might be worth to think of a better more error tolerant fallback if the folders couldn't be created. Maybe we just return the directory anyways and only keep the log?